### PR TITLE
Signup improvements

### PR DESCRIPTION
**Pull request by coreymunn3**
Several changes to Signup Including
- converted to functional component to avoid this binding
- used local state with useState hook for locations, loading, error, and formState (called 'newUser')
- fetching locations from API on component mount and using them to populate the select dropdown menu
- improved handling of onchange, and onsubmit events.
- performing form validation in onSubmit
- showing toasts containing error messages when submission fails for some reason
- re-routing to feed when signup successful

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
 import './App.global.scss';
 
 const App = () => {
-  return <Signup></Signup>;
+  return (
+    <Router>
+      <Switch>
+        <Route exact path='/signup' component={Signup} />
+        <Route exact path='/login' component={Login} />
+        <Route exact path='/' component={Feed} />
```
Comment (omane101) [inline client/src/App.js]: looks good but i feel like we should reserve '/' for initial landing page (when users aren't signed in or on a log in page). path='/home' makes more sense to me at the moment

File: client/src/login-component/index.js [hunk 2]
```diff
@@ -1,87 +1,67 @@
...
+                  variant='primary'
+                  type='submit'
                   onClick={() => handleSubmit()}
                 >
                   Submit
                 </Button>
               </Form.Group>
               <Form.Group>
-                <Button variant="secondary">Register</Button>
+                <Button variant='secondary'>Register</Button>
```
Comment (omane101) [inline client/src/login-component/index.js]: I am not able to go to sign up page from login page at the moment. Register button is static

File: client/src/login-component/index.js [hunk 3]
```diff
@@ -1,87 +1,67 @@
...
-  //         </Post>
-  //     );
-
-  //   };
+const Login = () => {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
 
   const handleSubmit = () => {
     // TODO: handle transition from login to post screen
```
Comment (omane101) [inline client/src/login-component/index.js]: As part of the routing enhancement, could you also had a transition here to feed page? I used `Links` in an earlier provision and worked smoothly

File: client/src/signup-component/index.js [hunk 4]
```diff
@@ -1,174 +1,187 @@
...
+// util
+import validateSignup from '../utils/validateSignup';
+
+const Signup = () => {
+  const history = useHistory();
+  // pull list of locations when component mounts
+  const [locations, setLocations] = useState();
+  useEffect(() => {
+    const getLocations = async () => {
+      const { data } = await axios.get('/api/locations');
```
Comment (omane101) [inline client/src/signup-component/index.js:22]: Nice, definitely more elegant than hard-coding

File: client/src/signup-component/index.js [hunk 5]
```diff
@@ -1,174 +1,187 @@
...
+                {locations &&
+                  locations.map(({ _id, city, state }) => (
+                    <option key={_id} value={_id}>{`${city}, ${state}`}</option>
+                  ))}
+              </Form.Control>
+            </Form.Group>
+          </Form.Row>
+
+          <Button variant='primary' type='submit' block>
+            {loading ? 'Processing...' : 'Sign Up'}
```
Comment (omane101) [inline client/src/signup-component/index.js:179]: From UI perspective, this is a nice addition

**Review by omane101 — CHANGES_REQUESTED** [review]

File: client/src/login-component/index.js (hunk 3 above)
Comment (omane101) [reply client/src/login-component/index.js]: or how you did it in signup

File: client/src/login-component/index.js (hunk 2 above)
Comment (coreymunn3) [reply client/src/login-component/index.js]: Hey, I didn't do any real work on the login component. Whatever is in Login is what was from your original commit.

File: client/src/login-component/index.js (hunk 3 above)
Comment (coreymunn3) [reply client/src/login-component/index.js]: Yeah I'll use links.

**Pull request closed without merging**